esp/platform/tokenserialization.hpp

```diff
@@ -51,4 +60,390 @@ class TokenSerializer
+        long double tmp = strtold(buffer, &end);
+        DeserializationResult result = Deserialization_UNKNOWN;
+
+        if (0 == tmp && end == buffer)
+        {
+            result = Deserialization_NOT_A_NUMBER;
+        }
+        else if (0 == tmp && ERANGE == errno)
+        {
+            result = Deserialization_UNDERFLOW;
+        }
+        else if ((-HUGE_VALL == tmp || HUGE_VALL == tmp) && ERANGE == errno)
+        {
+            result = Deserialization_OVERFLOW;
+        }
+        else if (!isEmptyOrWhitespace(end))
+        {
+            result = Deserialization_NOT_A_NUMBER;
+        }
+#if defined(_WIN32)
+// VS2015 generates this warning (as an error) with unsigned integral types,
+// even though this method does not process integral types.
+#pragma warning(suppress:4146)
+#endif
+        else if (tmp < -std::numeric_limits<TValue>::max())
+        {
+            result = Deserialization_OVERFLOW;
+        }
+        else if (tmp > std::numeric_limits<TValue>::max())
+        {
+            result = Deserialization_OVERFLOW;
+        }
+        else
+        {
+            value = TValue(tmp);
+            result = Deserialization_SUCCESS;
+        }
+
+        return result;
+    }
+    template <typename TValue>
+    void logResult(const char* buffer, DeserializationResult result) const
+    {
+        bool success = false;
+        if (Deserialization_SUCCESS == result)
+#if !defined(_DEBUG)
+            return;
+#else
+            success = true;
+#endif
+
+        const char* typeStr = typeid(TValue).name();
+        const char* resultStr = NULL;
+
+        switch (result)
+        {
+        case Deserialization_UNKNOWN:       resultStr = "unknown"; break;
+        case Deserialization_SUCCESS:       resultStr = "success"; break;
+        case Deserialization_BAD_TYPE:      resultStr = "bad type"; break;
+        case Deserialization_UNSUPPORTED:   resultStr = "unsupported"; break;
+        case Deserialization_INVALID_TOKEN: resultStr = "invalid token"; break;
+        case Deserialization_NOT_A_NUMBER:  resultStr = "not a number"; break;
+        case Deserialization_OVERFLOW:      resultStr = "overflow"; break;
+        case Deserialization_UNDERFLOW:     resultStr = "underflow"; break;
+        default:                            resultStr = "unexpected"; break;
+        }
+
+        if (success)
+            DBGLOG("Result of deserializing '%s' to type '%s': %s", buffer, typeStr, resultStr);
+        else
+            ERRLOG("Result of deserializing '%s' to type '%s': %s", buffer, typeStr, resultStr);
+    }
+    bool isEmptyOrWhitespace(const char* buffer) const
+    {
+        skipWhitespace(buffer);
+        return (!*buffer);
+    }
+    void skipWhitespace(const char*& buffer) const
+    {
+        while (isspace(*buffer)) buffer++;
+    }
+};
+
+// The template methods in TokenSerializer must be redefined by each subclass
+// to dispatch requests correctly.
+#define EXTEND_TOKENDESERIALIZER(base) \
```

Comment:
It might be more difficult to miss this if it's defined at the top of the header